src/Language/PureScript/Bundle.hs

```diff
@@ -308,12 +308,21 @@ withDeps (Module modulePath fn es) = Module modulePath fn (map expandDeps es)
       -- ^ exports.foo means there's a dependency on the public member "foo" of
       -- this module.
       = ([(m, nm, Public)], bn)
+    toReference (JSObjectLiteral _ props _) bn
+      = let
+          shorthandNames = mapMaybe unPropertyIdentRef (trailingCommaList props)
+        in
+          (map (\name -> (m, name, Internal)) shorthandNames, bn)
```

Comment:
I think this should also be a `mapMaybe` conditional on `name` being an element of `bn`; `foo` should not be exported in the below case:

```js
var foo = "foo";
exports.bar = function (foo) { return { foo }; };
```

Reply:
Oh yes of course, thanks!

Reply:
How does it look now?